leastsquare.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+
+
+def calculate_weight(basis_function, output_labels):
+
+    # For MW = S
+    S = np.dot(basis_function, output_labels)
```

Comment:
Another way to do this is as below, not sure which is more pythonic:
```python
S = basis_function@output_labels
```

Reply:
Simplified implementation. Implemented